src/libs/ReportUtils.js

```diff
@@ -1250,6 +1250,50 @@ function buildOptimisticAddCommentReportAction(text, file) {
     };
 }
 
+/**
+ * update optimistic parent reportAction when a comment is added or remove in the child report
+ * @param {String} parentReportAction - Parent report action of the child report
+ * @param {String} lastVisibleActionCreated - Last visible action created of the child report
+ * @param {String} type - The type of action in the child report
+ * @returns {Object}
+ */
+
+function updateOptimisticParentReportAction(parentReportAction, lastVisibleActionCreated, type) {
+    let childVisibleActionCount = parentReportAction.childVisibleActionCount || 0;
+    let childCommenterCount = parentReportAction.childCommenterCount || 0;
+    let childOldestFourAccountIDs = parentReportAction.childOldestFourAccountIDs;
+
+    if (type === CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD) {
+        childVisibleActionCount += 1;
+        const oldestFourAccountIDs = childOldestFourAccountIDs ? childOldestFourAccountIDs.split(',') : [];
+        if (oldestFourAccountIDs.length < 4) {
+            const index = _.findIndex(oldestFourAccountIDs, (accountID) => accountID === currentUserAccountID.toString());
+            if (index === -1) {
+                childCommenterCount += 1;
```

Comment:
I'm confused why we are adding a commenter based on the `oldestFourAccountIDs`. We want to check that they've never commented in the thread before changing this right?

Reply:
I think this should still be extracted outside of the `<4` section actually. With the code you have here, we will _only_ increase the `childCommenterCount` if there are less than 4. We need to increase that regardless of how many people have commented, if it is this user's first time commenting. Right?